source/Octopus.Versioning/Maven/MavenPackageID.cs

```diff
@@ -24,6 +24,30 @@ namespace Octopus.Versioning.Maven
     /// </summary>
     public class MavenPackageID
     {
+        /// <summary>
+        /// Standard GAV coordinates are group:artifact:version. This can also be extended to include the packaging in the
+        /// format group:artifact:packaging:version.
+        ///
+        /// However, we never pass the version in the package id from the UI. Instead we pass a string like group:artifact or
+        /// group:artifact:packaging. This is because the version selection is a separate process from the package definition.
+        ///
+        /// It is here that we convert the package id sent from the UI into the standard Maven string.
+        /// </summary>
+        /// <param name="input">The input sent from the UI</param>
+        /// <param name="version">The optional version</param>
+        /// <returns>A MavenPackageID created to match the package id an optional packaging defined in the UI</returns>
+        /// <exception cref="ArgumentException">thrown if the input is not in the correct format</exception>
+        public static MavenPackageID CreatePackageIDFromUIInput(string input, IVersion version = null)
```

Comment:
This method doesn't feel quite right.

It makes the IDE angry because it doesn't follow the camel-case naming standard, but it's not that.
I don't think this library should refer to the UI.  The dependency is the other way around.  If the code is UI specific, then it should live in Octopus.Server.
But also, when would I call this method vs calling the MavenPackageId(string, Version) constructor?
Maybe this could be combined with the constructor?

Reply:
The issue is that this code is called from both calamari and octopus, so it is either copied and pasted in both projects or lives here.

Reply:
The reason for this method being "outside" of the main class was to separate the Octopus specific implementation from what was otherwise generic code that is bound to the maven implementation. 

The syntax for the package coordinates are defined in http://maven.apache.org/plugins/maven-dependency-plugin/get-mojo.html as `groupId:artifactId:version[:packaging[:classifier]]`, which defines that the coordinates for a specific form of packaging is appended after the version. The method explicitly calls out that we need to bend these rules a little to not include the version, and so are effectively creating our own syntax.